specification/eventgrid/data-plane/Microsoft.EventGrid/preview/2023-06-01-preview/EventGrid.json

```diff
@@ -672,6 +672,146 @@
         "failedLockTokens",
         "succeededLockTokens"
       ]
+    },
+    "EventGridNamespaceMQTTClientCreatedOrUpdatedEventData": {
+      "type": "object",
+      "description": "Event data for Microsoft.EventGrid.MQTTClientCreatedOrUpdated event.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/MQTTClientCreatedOrUpdatedEventProperties"
+        }
+      ],
+      "properties": {}
+    },
+    "EventGridNamespaceMQTTClientDeletedEventData": {
+      "type": "object",
+      "description": "Event data for Microsoft.EventGrid.MQTTClientDeleted event.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/MQTTClientDeletedEventProperties"
+        }
+      ],
+      "properties": {}
+    },
+    "EventGridNamespaceMQTTClientSessionConnectedEventData": {
+      "type": "object",
+      "description": "Event data for Microsoft.EventGrid.MQTTClientSessionConnected event.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/MQTTClientSessionConnectedEventProperties"
+        }
+      ],
+      "properties": {}
+    },
+    "EventGridNamespaceMQTTClientSessionDisconnectedEventData": {
+      "type": "object",
+      "description": "Event data for Microsoft.EventGrid.MQTTClientSessionDisconnected event.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/MQTTClientSessionDisconnectedEventProperties"
+        }
+      ],
+      "properties": {}
+    },
+    "MQTTClientCreatedOrUpdatedEventProperties": {
+      "type": "object",
+      "description": "Schema of the Data property of an EventGridEvent for MQTT Client Created Or Updated event.",
+      "properties": {
+        "clientAuthenticationName": {
+          "type": "string",
+          "description": "Unique identifier for the MQTT client that the client presents to the service for authentication. This case-sensitive string can be up to 128 characters long, and supports UTF-8 characters."
+        },
+        "clientName": {
+          "type": "string",
+          "description": "Name of the client resource in the Event Grid namespace."
+        },
+        "namespaceName": {
+          "type": "string",
+          "description": "Name of the Event Grid namespace where the MQTT client was created or updated."
+        },
+        "state": {
+          "type": "string",
+          "description": "Configured state of the client. The value could be Enabled or Disabled."
+        },
+        "createdOn": {
+          "type": "string",
+          "description": "Time the client resource is created based on the provider's UTC time."
+        },
+        "updatedOn": {
+          "type": "string",
+          "description": "Time the client resource is last updated based on the provider's UTC time. If the client resource was never updated, this value is identical to the value of the 'createdOn' property."
+        },
+        "attributes": {
```

Comment:
Is this supposed to be a dictionary? You would need to add additionalProperties along with the type of the value, e.g. https://github.com/Azure/azure-rest-api-specs/blob/main/specification/eventgrid/data-plane/Microsoft.Media/stable/2018-01-01/MediaServices.json#L114-L116

Reply:
not sure what you mean with dictionary here.